### PR TITLE
Collection view Events

### PR DESCRIPTION
**Pull request by scott-w**
### Proposed changes
 - Rename marionette.events.md to events.md
 - Change the CollectionView events to reflect View events structure

Link to the issue: #3052

### CODE REVIEW

File: docs/marionette.collectionview.md [hunk 1]
```diff
@@ -977,67 +979,144 @@ var MyCollectionView = Mn.CollectionView.extend({
...
+[Documentation for Child View Events](./events.md#child-view-events) for more
+information.
+
+### Lifecycle Events
+
+The `CollectionView` has its own lifecycle around the standard `View` event
+rendering lifecycle. This section covers the events that get triggered and what
+they indicate.
+
+#### View Creation Lifecycle
```
Comment (denar90) [inline docs/marionette.collectionview.md]: Don't you mind to use `CollectionView` instead of `View`?

File: docs/marionette.collectionview.md [hunk 2]
```diff
@@ -30,31 +29,37 @@ in the DOM. This behavior can be disabled by specifying `{sort: false}` on initi
...
   * [CollectionView's `getViewComparator`](#collectionviews-getviewcomparator)
   * [CollectionView's `reorderOnSort`](#collectionviews-reorderonsort)
   [CollectionView's `reorder`](#collectionviews-reorder)
   * [CollectionView's `resortView`](#collectionviews-resortview)
-
-* [CollectionView Events](#collectionview-events)
+* [Events](#collectionview-events)
+  * [Child Event Bubbling](#child-event-bubbling)
+  * [Lifecycle Events](#lifecycle-events)
+    * [View Creation Lifecycle](#view-creation-lifecycle)
```
Comment (ahumphreys87) [inline docs/marionette.collectionview.md]: maybe drop `View` from here, its not used in any of the events headings.

File: docs/marionette.collectionview.md [hunk 3]
```diff
@@ -30,31 +29,37 @@ in the DOM. This behavior can be disabled by specifying `{sort: false}` on initi
...
+      * [Collection `before:render`](#collection-beforerender)
+      * [Collection `before:render:empty`](#collection-before-renderempty)
+      * [Collection `before:render:children`](#collection-before-renderchildren)
+      * [Collection `before:add:child`](#collection-beforeaddchild)
+      * [Collection `add:child`](#collection-addchild)
+      * [Collection `render:empty`](#collection-renderempty)
+      * [Collection `render:children`](#collection-renderchildren)
+      * [Collection `render`](#collection-render)
+    * [View Destruction Lifecycle](#view-destruction-lifecycle)
```
Comment (ahumphreys87) [inline docs/marionette.collectionview.md]: same, maybe drop `View`?

File: docs/marionette.collectionview.md (hunk 1 above)
Comment (ahumphreys87) [inline docs/marionette.collectionview.md]: I think drop the `View` altogether.

File: docs/marionette.collectionview.md [hunk 4]
```diff
@@ -977,67 +979,144 @@ var MyCollectionView = Mn.CollectionView.extend({
...
+
+### Lifecycle Events
+
+The `CollectionView` has its own lifecycle around the standard `View` event
+rendering lifecycle. This section covers the events that get triggered and what
+they indicate.
+
+#### View Creation Lifecycle
+
+The `CollectionView` creation lifecycle can go down two paths depending on
```
Comment (ahumphreys87) [inline docs/marionette.collectionview.md]: again, remove `CollectionView`

File: docs/marionette.collectionview.md [hunk 5]
```diff
@@ -977,67 +979,144 @@ var MyCollectionView = Mn.CollectionView.extend({
...
+|  2*   |  `before:render:empty`   |
+|  2+   | `before:render:children` |
+|   3   |    `before:add:child`    |
+|   4   |       `add:child`        |
+|  5*   |      `render:empty`      |
+|  5+   |     `render:children`    |
+|   6   |          `render`        |
+|   7   |      `before:attach`     |
+|   8   |         `attach`         |
+|   9   |       `dom:refresh`      |
```
Comment (ahumphreys87) [inline docs/marionette.collectionview.md]: I love this btw 👍

File: docs/marionette.collectionview.md [hunk 6]
```diff
@@ -977,67 +979,144 @@ var MyCollectionView = Mn.CollectionView.extend({
...
-
 myView.render();
 ```
 
+##### Collection `render`
+
+Fires when the collection has completely finished rendering. See the
+[`render` Documentation](./marionette.view.md#view-render) for more information.
+
+#### View Destruction Lifecycle
```
Comment (ahumphreys87) [inline docs/marionette.collectionview.md]: drop `View`

File: docs/marionette.collectionview.md [hunk 7]
```diff
@@ -977,67 +979,144 @@ var MyCollectionView = Mn.CollectionView.extend({
...
+
+When a `CollectionView` is destroyed it fires a series of events in order to
```
Comment (ahumphreys87) [inline docs/marionette.collectionview.md]: drop `CollectionView`

**Comment by ahumphreys87** [comment]
👍

**Comment by denar90** [comment]
👍

**Pull request merged**